zjit/src/codegen.rs

```diff
@@ -379,6 +381,53 @@ fn gen_putspecialobject(asm: &mut Assembler, value_type: SpecialObjectType) -> O
     )
 }
 
+fn gen_getspecial_symbol(asm: &mut Assembler, _key: u64, svar: u64) -> Opnd {
+    // Fetch a "special" backref based on a char encoded by shifting by 1
+
+    // call rb_backref_get()
+    asm_comment!(asm, "rb_backref_get");
+    let backref = asm.ccall(rb_backref_get as *const u8, vec![]);
+
+    let rt_u8: u8 = (svar >> 1).try_into().unwrap();
+    match rt_u8.into() {
+        '&' => {
+            asm_comment!(asm, "rb_reg_last_match");
+            asm.ccall(rb_reg_last_match as *const u8, vec![backref])
+        }
+        '`' => {
+            asm_comment!(asm, "rb_reg_match_pre");
+            asm.ccall(rb_reg_match_pre as *const u8, vec![backref])
+        }
+        '\'' => {
+            asm_comment!(asm, "rb_reg_match_post");
+            asm.ccall(rb_reg_match_post as *const u8, vec![backref])
+        }
+        '+' => {
+            asm_comment!(asm, "rb_reg_match_last");
+            asm.ccall(rb_reg_match_last as *const u8, vec![backref])
+        }
+        _ => panic!("invalid back-ref"),
```

Comment:
```suggestion
        c => panic!("invalid back-ref {c}"),
```

Reply:
If you end up making an enum, it would be great to shift this into the parser into HIR so that we have more up-front guarantees of program legality